src/vagrant.rb

```diff
@@ -579,14 +579,15 @@ def configure
 
       trigger_actions = (File.exist?(host_file_path) && File.size(host_file_path) > 0) ? [:provision] : [:up, :provision]
 
-      FileUtils.mkdir_p File.dirname(host_file_path)
-      FileUtils.touch host_file_path
-
       machine.vagrant.trigger.before trigger_actions do |trigger|
         trigger.name = "#{name}_chef_install"
         trigger.run = {
           inline: "chef install #{policyfile_path}",
         }
+        trigger.ruby do
+          FileUtils.mkdir_p File.dirname(host_file_path)
+          FileUtils.rm_f host_file_path
```

Comment:
I think this was at the original place for and used touch for a reason. If you delete the .chef folder after the machine was created, Vagrant will report errors for that machine as the file provisioner will not find the file.

Reply:
Great point! I did see that issue, but if you look at the next triggers right below, that folder will be created with the cookbooks and the compressed file. You could say that I simply moved upwards the `rm` command in the other line I changed (603).

Reply:
It's not about the `rm`, it's clear that was simply moved, but that the original `touch` call is missing.

You can test this in the `samples\provisioners\core\linux` folder for example. To me the following throws an error there:

1. `vagrant up`
1. Remove the `.chef` folder
1. `vagrant status`

That's the reason I had the `touch` there earlier. I know it's probably an edge case, but this worked with the original version.

Anyhow, my other comment also applies, you can temporarily simply ignore this custom provisioner at the packer repo, I had plans to get rid of this dependency there anyway.

Reply:
Well, what I meant was that I didn't change the proper functionality. The subsequent triggers right below this one will still create the `.chef` folder and populate it with the cookbook folder and compressed file. I didn't find anything deleting that directory afterwards. If I understood everything correctly, explicitly touching the file is unnecessary in that case.

Anyways, if this isn't necessary, no worries. I'll close this PR and open one for `packer`. 🙂

Reply:
What I meant was if you _manually_ delete the `.chef` folder, then subsequent Vagrant commands will fail. You are right that this is probably not the most common scenario, similar to someone deleting the `.vagrant` folder itself, but this was the reason I originally included the `touch` call.

Maybe a bettter approach would be to create the `.chef` folder inside the `.vagrant` one to have less temporary files, or simply getting rid of the zipping whatsoever. I originally chose this approach as some file provisioners (e.g. over WinRM) get really slow when uploading multiple files, but SSH is now widely supported on Windows as well, and that does not have this limitation.

Thanks again for your efforts, looking forward to the `packer` improvements ;) I'll look into simplifying this part anyway soon.